pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -749,68 +755,94 @@ func (og *operationGenerator) GenerateVerifyControllerAttachedVolumeFunc(
 			return nil
 		}
 
-		if !volumeToMount.ReportedInUse {
-			// If the given volume has not yet been added to the list of
-			// VolumesInUse in the node's volume status, do not proceed, return
-			// error. Caller will log and retry. The node status is updated
-			// periodically by kubelet, so it may take as much as 10 seconds
-			// before this clears.
-			// Issue #28141 to enable on demand status updates.
-			return fmt.Errorf("Volume %q (spec.Name: %q) pod %q (UID: %q) has not yet been added to the list of VolumesInUse in the node's volume status",
-				volumeToMount.VolumeName,
-				volumeToMount.VolumeSpec.Name(),
-				volumeToMount.PodName,
-				volumeToMount.Pod.UID)
-		}
-
-		// Fetch current node object
-		node, fetchErr := og.kubeClient.Core().Nodes().Get(string(nodeName), metav1.GetOptions{})
-		if fetchErr != nil {
+		timeout := int64(nodeWatchTimeoutInSeconds)
+		nodeSelector := fields.OneTermEqualSelector("metadata.name", string(nodeName))
+		w, err := og.kubeClient.Core().Nodes().Watch(metav1.ListOptions{
+			FieldSelector:   nodeSelector.String(),
+			ResourceVersion: "0",
+			TimeoutSeconds:  &timeout,
+		})
```

Comment:
Your are right, it triggers on all changes to the node status. The only filtering that is done is with the nodeSelector to only watch the desired node.

The watch will not stop before the timeout is over. Due to the "0" for ResourceVersion, it will immediately give the current node status without waiting and then keep giving new statuses until the timeout is over. I have this in my custom built 1.5.x build and deployed since a few days and it works as expected.

I had problems the ReportedInUse check as it bailed out immediately. It used the kubeletes node status cache while at the same time the (old) Get call that came after the check, used more recent information. I decided to make both checks rely on the result from the watch to 1. fix the early bail out and 2. fix another possible delay of 10 seconds.
After this PR is merged, I'll probably remove the ReportedInUse and MarkVolumesAsReportedInUse logic as the information was only used in this place. Are you ok with that @saad-ali (As I see you introduced the logic)

Btw...while verifying the code again, I noticed that I forgot to close the watch in case VolumesAttached reported attach pretty early. I added a "defer w.Stop()" now.